### ISSUE TITLE
crash on existing file load

### ISSUE THREAD

**Issue by epsimpson** [CleverRaven/Cataclysm-DDA#52358]
### Describe the bug

Game crashes while loading save from version 0.F-3357-g95019f2a75 into 0.F-3460-g157f88ac1e.

### Steps To Reproduce

1. Start version 0.F-3460-g157f88ac1e
2. Attempt to load character "Katharyn Linn" provided save file.

### Expected behavior

save file loads and runs.

### Versions and configuration

- OS: Linux
    - OS Version: Distributor ID: Ubuntu; Description: Ubuntu 20.04.3 LTS; Release: 20.04; Codename: focal; 
- Game Version: 0.F-3460-g157f88ac1e [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    No Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions]
]

### Additional context

There is no /config/crash.log file.
I created a second character to ensure a new character would generate properly.

[debug.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7375079/debug.zip)

[AlexLoadCrash.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7375094/AlexLoadCrash.zip)

**Comment by estebandellasilva** [comment]
Probably related to https://github.com/CleverRaven/Cataclysm-DDA/pull/51932

which removed the acid from batteries recipe, but after trying the remove the recipe from my character it still crashes, but without any information why

**Comment by ZhilkinSerg** [comment]
![image](https://user-images.githubusercontent.com/16213433/137979559-238c4766-f1f5-4b5a-907c-a51d09c8e00c.png)

**Comment by ProfoundDarkness** [comment]
This looks a LOT like what I'm having today.  On Linux it just ends the process (no crash log).  On windows (MXE compile) it crashes (crash log, even though compiling with libbacktrace it's just memory addresses).  I started a bisect earlier but couldn't finish it until now... results:  8b2d5642d6f63cb4ed8c162f41d62ab010f2792e is the first bad commit

I edited my save to remove all references to the eocs: "night_messages", "night_messages2", "night_messages3" and my save loaded fine...  That was rather tedious however as the text editor doesn't like nearly 10 meg files and apparently I don't remember the non-greedy regex syntax...

Anyway, post save edit I reset the bisect and I moved up to master I pulled this morning,  version 157f88ac1e and my save loads fine still.  I'd swear I've seen another instance recently of eocs being changed resulting in crashes from loading saves.

Just for the heck of it: 
[AlexLoadCrash-fixed.zip](https://github.com/CleverRaven/Cataclysm-DDA/files/7377530/AlexLoadCrash-fixed.zip)
